test/unit/product/product-view.js

```diff
@@ -54,40 +55,210 @@ describe('Product View class', () => {
     document.body.removeChild(configCopy.node);
   });
 
+  describe('resizeOnLoad', () => {
+    let resizeStub;
+    let addEventListenerStub;
+    let wrapper;
+    let node;
+
+    beforeEach(() => {
+      resizeStub = sinon.stub(View.prototype, 'resize');
+      addEventListenerStub = sinon.stub(EventTarget.prototype, 'addEventListener');
+      product = Object.defineProperty(product, 'classes', {
+        value: {
+          product: {
+            img: 'img-class',
+          },
+        },
+      });
+      wrapper = document.createElement('div');
+      node = document.createElement('div');
+      wrapper.appendChild(node);
+      product.view.wrapper = wrapper;
+    });
+
+    afterEach(() => {
+      resizeStub.restore();
+      addEventListenerStub.restore();
+    });
+
+    it('does not add event listener if product contents do not have an image or image with carousel', () => {
+      product.config.product.contents = {
+        img: null,
+        imgWithCarousel: null,
+      };
+      product.view.resizeOnLoad();
+      assert.notCalled(addEventListenerStub);
+    });
+
+    it('does not add event listener if there is no image in the wrapper', () => {
+      product.config.product.contents = {
```

Comment:
For clarity, could we move the instantiation of `node` and `wrapper.appendChild` out of the `beforeEach` and into the tests below? We could explicitly add classes to `node` which either match or don't match `img-class`.